src/Parsers/ASTColumnDeclaration.cpp

```diff
@@ -78,7 +79,7 @@ void ASTColumnDeclaration::formatImpl(const FormatSettings & settings, FormatSta
     if (default_expression)
     {
         settings.ostr << ' ' << (settings.hilite ? hilite_keyword : "") << default_specifier << (settings.hilite ? hilite_none : "");
-        if (default_specifier != "EPHEMERAL" || !default_expression->as<ASTLiteral>()->value.isNull())
+        if (!ephemeral_default)
```

Comment:
I'm a bit confused why we can't have
```suggestion
        if (default_specified != "EPHEMERAL")
```
because this `ephemeral_default` represents a presence of default_expression for `EPHEMERAL` default_specifier

Reply:
we have special meaning for `EPHEMERAL` different from `DEFAULT` - `EPHEMERAL` signifies that column is ephemeral, and then it may have optional expression - if this expression present - it's a default value, if not - default is a default value of column's type - so, the flag `ephemeral_default` means that this expression is not present (and value is defaulted to type default) - which means we don't want to print it out at all. However if `EPHEMERAL` contains expression, then `ephemeral_default` is false and we want to print this expression. For default_specifier `DEFAULT` expression is always present, `ephemeral_default` is not changed and always false, and we always want to print this expression - so, from `ephemeral_default` we always know if we want to print this expression